Update tutorial notebook, document issues with inference

jybook opened the pull request:
There's a known issue with pymc3 ignoring bounds on priors and then ending up in impossible places. (similar to the open issues documented here: https://github.com/ICB-DCM/pyPESTO/issues/365 and here: https://github.com/pymc-devs/pymc3/issues/4279, and here: https://discourse.pymc.io/t/square-decision-region-model-specification-initial-evaluation-failed/6413

We discovered this bug at around 3am, and will continue attempting to fix it, but would appreciate help from the teaching staff is possible.

Otherwise, we can rewrite the model before the next deadline.

jybook committed:
Update tutorial notebook, document issues with inference

There's a known issue with pymc3 ignoring bounds on priors and then ending up in impossible places. (similar to the open issues documented here: https://github.com/ICB-DCM/pyPESTO/issues/365 and here: https://github.com/pymc-devs/pymc3/issues/4279, and here: https://discourse.pymc.io/t/square-decision-region-model-specification-initial-evaluation-failed/6413

We discovered this bug at around 3am, and will continue attempting to fix it, but would appreciate help from the teaching staff is possible.

Otherwise, we can rewrite the model before the next deadline.

jybook committed:
rename inference_functions to module as specified in assignment

jybook committed:
Merge branch 'main' into inference_code

jybook committed:
Switches fit method to mcmc, adds tests

Also does a bunch of other stuff in terms of cleaning up code, minor updates to stat model, but nothing else too significant

jybook committed:
change likelihood to poisson, partial update of tutorial

jybook committed:
Still a work in progress, just cleaned up a bit

Why so many divergences?

tcontrer commented:
test_inference.py does not load packages correctly

- Resolved by deleting test_inference.py, which was made obsolete by test_module.py

tcontrer commented:
test_module.py takes way too long to run. Can we run it with just simulating 10 neutrinos? 
- Have asked @jennmcguire for feedback here

tcontrer commented:
Should 'test_accepts_resonable_inputs' to something like 'test_rejects_unreasonable_inputs'
- done!

tcontrer commented on bayboone/inference/module.py:
```diff
@@ -0,0 +1,132 @@
+        
+    return osc_model
+
+def fit_model(data):
```
Can we keep this consistent with the rest of the package and use N_numu, N_nue, rather that this data array?

jybook replied:
yup

tcontrer commented on bayboone/tests/test_module.py:
```diff
@@ -0,0 +1,37 @@
+        #test_model = module.oscillation_model(num_numu, num_nue)       
+        self.assertRaises(ValueError, module.oscillation_model, num_numu, num_nue)    
+        
+class TestInference(TestCase):
```
data_io no longer has generative_model, since it was never a generative model. It only simulates data, which I test in test_io.py, so you can get rid of this here.

tcontrer commented:
Update TestInference_functions to match new edits in data_io.py and module.py
- Resolved by deleting test_inference.py, which was made obsolete by test_module.py

tcontrer committed:
Updated tutorial notebook

jybook committed:
edits before due date #2

Mostly addressing direct feedback from comments and pull requests, some more tuning of generative model

jybook committed:
tried clearing cells in tutorial notebook

jybook committed:
hopefully this works...
Merge branch 'main' into inference_code
